src/bsp/src/stm32/src/SocketFactory.cpp

```diff
@@ -23,6 +23,7 @@ std::optional<TCPClient> SocketFactory::createTCPClient(const char* address,
     // Connect the socket
     if (lwip_connect(sockfd, (sockaddr*)&serverAddr, sizeof(serverAddr)) < 0) {
         logger.log(LogLevel::Info, "Could not connect to server %s : %d", address, port);
+        lwip_close(sockfd);
```

Comment:
Do we need to do the same thing on the POSIX version?

Reply:
hmmmm good question, I'll check it out

Reply:
Posix socket are now closed on fail